pkg/cmd/release/create/gitResources.go

```diff
@@ -0,0 +1,301 @@
+		default:
+			if resolvedOverride != nil {
+				overriddenGitResourceGitRefs = append(overriddenGitResourceGitRefs, resolvedOverride)
+				gitResourceGitRefsWithOverrides = ApplyGitResourceOverrides(gitResourcesBaseline, overriddenGitResourceGitRefs)
+			}
+		}
+		// loop around and let them put in more input
+	}
+	return overriddenGitResourceGitRefs, nil
+}
+
+func ParseGitResourceGitRefString(gitResourceRef string) (*GitResourceGitRef, error) {
+	if strings.TrimSpace(gitResourceRef) == "" {
+		return nil, errors.New("empty git resource git ref specification")
+	}
+
+	components := splitString(gitResourceRef, []int32{':', '='})
+	actionName, gitResourceName, gitRef := "", "", ""
+
+	// We support 2 formats of git resource
+	// {StepName}:{GitRef} - This targets the primary git resource for a step only
+	// {StepName}:{GitResourceName}:{GitRef} - This targets a secondary git resource
+
+	actionName = strings.TrimSpace(components[0])
+	switch len(components) {
+	case 2:
+		gitRef = strings.TrimSpace(components[1])
+	case 3:
+		gitResourceName = strings.TrimSpace(components[1])
+		gitRef = strings.TrimSpace(components[2])
+	default:
+		return nil, fmt.Errorf("git resource git ref specification \"%s\" does not use expected format", gitResourceRef)
+	}
+
+	if actionName == "" {
+		return nil, fmt.Errorf("git resource git ref specification \"%s\" cannot have an empty step name", gitResourceRef)
+	}
+
+	if gitRef == "" {
+		return nil, fmt.Errorf("git resource git ref specification \"%s\" cannot have an empty git ref", gitResourceRef)
+	}
+
+	return &GitResourceGitRef{
+		ActionName:      actionName,
+		GitRef:          gitRef,
+		GitResourceName: gitResourceName,
+	}, nil
+}
+
+func ResolveGitResourceOverride(override *GitResourceGitRef, baseline []*GitResourceGitRef) (*GitResourceGitRef, error) {
+	index := slices.IndexFunc(baseline, func(grgr *GitResourceGitRef) bool {
+		return grgr.Equals(override)
+	})
+
+	if index < 0 {
+		return nil, fmt.Errorf("could not resolve step name \"%s\" or git resource name \"%s\"", override.ActionName, override.GitResourceName)
+	}
+
+	//If the user override matches a git resource in the "baseline"
+	var gitRef string
+	if override.GitRef == "*" {
+		gitRef = baseline[index].GitRef
+	} else {
+		gitRef = override.GitRef
+	}
+
+	return &GitResourceGitRef{
+		ActionName:      override.ActionName,
+		GitRef:          gitRef,
+		GitResourceName: override.GitResourceName,
+	}, nil
+}
+
+func ApplyGitResourceOverrides(gitResourcesBaseline []*GitResourceGitRef, overrides []*GitResourceGitRef) []*GitResourceGitRef {
+	result := make([]*GitResourceGitRef, 0, len(gitResourcesBaseline))
+
+	for _, gitResourceGitRef := range gitResourcesBaseline {
+
+		//Try and find any override
+		overrideIdx := slices.IndexFunc(overrides, func(override *GitResourceGitRef) bool {
+			return override.Equals(gitResourceGitRef)
+		})
```

Comment:
I may have misunderstood how the overrides work, but aren't they a stack? So if we find the first match here, could we be missing more recent overrides for the same ref?

There doesn't seem to be a test for this scenario that I can see either.

Reply:
oo, good point. Let me check that

Reply:
Fixed in 3f90de7